openmodelica_microgrid_gym/execution/runner.py

```diff
@@ -48,14 +59,13 @@ def run(self, n_episodes: int = 10, visualise: bool = False):
                 if done:
                     break
             self.agent.observe(r, done)
-            _ , env_fig = self.env.close()
+            _, env_fig = self.env.close()
 
             if visualise:
-                self.agent.render()
-
-            if self.agent.has_improved:
-                #self.best_episode['best_agent_plt'] = self.agent.figure
-                self.best_episode['best_env_plt'] = env_fig
-                self.best_episode['best_episode_idx'] = self.agent.best_episode
+                agent_fig = self.agent.render()
 
+            self.best_episode['agent_plt'] = agent_fig
 
+            if i == 0 or self.agent.has_improved:
+                self.best_episode['best_env_plt'] = env_fig
+                self.best_episode['i'] = i
```

Comment:
Why is the agent fig always updated and not only if there agent improved

Reply:
Due to the last agent plot has to be plotted.
dict name and key names changed to make it more clear